packages/core/src/persistent-storage/user-choices.ts

```diff
@@ -0,0 +1,92 @@
+import { cssPrefix } from '../constants';
+import { getLocalStorageObject, setLocalStorageObject } from './local-storage-helpers';
+
+const USER_CHOICES_KEY = `${cssPrefix}-device-settings` as const;
+
+/**
+ * Represents the user's choices for video and audio input devices,
+ * as well as their username.
+ */
+export type UserChoices = {
+  /**
+   * Whether video input is enabled.
+   * @defaultValue `true`
+   */
+  videoInputEnabled: boolean;
+  /**
+   * Whether audio input is enabled.
+   * @defaultValue `true`
+   */
+  audioInputEnabled: boolean;
+  /**
+   * The device ID of the video input device to use.
+   * @defaultValue `''`
+   */
+  videoInputDeviceId: string;
+  /**
+   * The device ID of the audio input device to use.
+   * @defaultValue `''`
+   */
+  audioInputDeviceId: string;
+  /**
+   * The username to use.
+   * @defaultValue `''`
+   */
+  username: string;
+};
+
+const defaultUserChoices: UserChoices = {
+  videoInputEnabled: true,
+  audioInputEnabled: true,
+  videoInputDeviceId: '',
+  audioInputDeviceId: '',
+  username: '',
+} as const;
+
+/**
+ * Saves user choices to local storage.
+ * @param deviceSettings - The device settings to be stored.
+ * @alpha
+ */
+export function saveUserChoices(
+  deviceSettings: UserChoices,
+  /**
+   * Whether to prevent saving user choices to local storage.
+   */
+  preventSave: boolean = false,
```

Comment:
could we instead of the boolean parameter simply _not_ call the function if the user doesn't intend to save?

Reply:
I wanted the default `defaultUserChoices` (if no defaults are passed by the user) to be defined and handled within the core. This API evolved from that decision. We could move the same logic one layer up into components-react, but that would increase the API surface a bit, and (if we support more frameworks in the future) we would have to reimplement it.

Reply:
got it, that makes sense to me!